tests/unit/hive_metastore/test_locations.py

```diff
@@ -718,3 +719,184 @@ def my_side_effect(path, **_):
     assert results[0].location == "adls://bucket/table1"
     assert results[1].location == "dbfs:/mnt/test_mount/table2"
     assert results[2].location is None
+
+
+def test_mount_listing_misplaced_flat_file():
```

Comment:
What should happen with the misplaced flat file? Should it be ignored by the tables in mounts?

Reply:
No, there is no simple heuristic to ignore it, so we file it as a table in case it's important.